app/src/main/java/edu/hm/itsec/taptrapattackshowcase/notificationservice/NotificationServiceAttackActivity.kt

```diff
@@ -0,0 +1,181 @@
+package edu.hm.itsec.taptrapattackshowcase.notificationservice
+
+import android.app.Activity
+import android.content.ComponentName
+import android.content.Context
+import android.content.Intent
+import android.database.ContentObserver
+import android.net.Uri
+import android.os.Bundle
+import android.os.CountDownTimer
+import android.os.Handler
+import android.os.Looper
+import android.provider.Settings
+import android.util.Log
+import android.view.View
+import android.widget.Toast
+import androidx.activity.enableEdgeToEdge
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.app.ActivityOptionsCompat
+import androidx.lifecycle.lifecycleScope
+import edu.hm.itsec.taptrapattackshowcase.R
+import edu.hm.itsec.taptrapattackshowcase.databinding.ActivityNotificationServiceAttackBinding
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+class NotificationServiceAttackActivity : AppCompatActivity() {
+
+    private val TAG = "NotificationServiceAttackActivity"
+
+    private lateinit var binding: ActivityNotificationServiceAttackBinding
+
+    private var transparencyDeactivated = false
+
+    private var granted: Boolean = false
+
+    private var resultLauncher = registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result ->
+        if (result.resultCode == Activity.RESULT_OK) {
+            granted = true
+        }
+    }
+
+    private val settingsObserver = object : ContentObserver(Handler(Looper.getMainLooper())) {
+        override fun onChange(selfChange: Boolean, uri: Uri?) {
+            super.onChange(selfChange, uri)
+
+            // check if permission is granted
+            if (notificationListenerServicePermissionGranted(applicationContext)) {
+                // bring app to front
+                val intent = Intent(this@NotificationServiceAttackActivity, NotificationServiceAttackActivity::class.java).apply {
+                    addFlags(Intent.FLAG_ACTIVITY_REORDER_TO_FRONT)
```

Comment:
```suggestion
                    addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_SINGLE_TOP)
```

Würde ich so machen, damit man beim Zurück-Navigieren sauber zur NotificationServiceActivity kommt.